### ISSUE TITLE
Note on clarity of terms: Objects, Stix Objects, Domain Objects, Marking Definitions, Relationships

### ISSUE THREAD

**Issue by StephenOTT** [oasis-tcs/cti-stix2#114]
There seems to be some unintentional overlap of terms when reading the docs that make it confusing

The Bundle docs read:

>A Bundle is a collection of arbitrary STIX Objects and Marking Definitions grouped together in a single container.

objects (optional) | list of type <STIX Object> or marking-definition | Specifies a set of one or more STIX Objects. Objects in this list MUST be a STIX Object (SDO, SRO or Custom Object) or a Marking Definition object.
-- | -- | --

----

Is there clarification around how to separate this overlapping use of "objects" and the individual objects?

The way i have been looking at it from an implementation stand point:

1. STIX Bundle
1. STIX Objects
    1. STIX Domain Object
    1. STIX Relationship Objects
        1. Relationship
        1. Sighting _"relationship"_
    1. STIX Data Marking (Data Marking Definition)

Basically meaning that there are Bundles and Objects.  Objects go into a Bundle.  Everything else is a type of object.

I look for this clarification because when a implementor is designing the data structure, this nuance is important for the structure that they will create.  The most upfront logic can be provided the better! :)

**Comment by jordan2175** [comment]
A STIX Bundle is just a transport wrapper and is not meant to be kept around.  So you have STIX Objects (SDOs and SROs) and other STIX things.  These other STIX things are things like Marking Definitions and the new Language thing for internationalization.  

Things the inherit from the "Common Properties" are STIX Objects.  Things that do not inherit are something else.

What would you suggest we do to make this more clear?  We have struggled with terminology for grouping and bucketing these types of things.

**Comment by StephenOTT** [comment]
My issue is from "objects" being seemingly overused in the spec docs.  Where a developer/implementer needs to constantly remind that STIX Objects are not all types of objects that are top level "objects" in a Bundle's "objects" property. (wrote that to be confusing on purpose to show the issue).

A Bundle is a wrapper, but defines itself as a wrapper for "objects".  And we are in the STIX namespace of "objects".


The clarification that I made for a implementation is as a wrote above:


1. STIX Bundle
1. STIX Objects
    1. STIX Domain Object
    1. STIX Relationship Objects
        1. Relationship
        1. Sighting "relationship"
    1. STIX Data Marking (Data Marking Definition)

Where Stix Objects are any object that are "Bundle Objects" (things that are significant enough to be managed as individual objects rather than embedded objects inside of other objects (such as Cyber Observables)).

A Marking Definition is a "Stix Object" because its a Top level item that is managed inside of the Wrapper (the bundle): it was "important enough" to transport around as a first class object rather than a embedded object

**Comment by jordan2175** [comment]
I fully understand the concerns you have.  We as editors have struggled with what to call all of these various layers without making it more complicated.  Keep in mind you are missing the Language object in your list.  It is not an SDO or SRO.  So there is more objects like the Marking Definition object that do NOT inherit the common properties.  

If you have any suggestions for terminology that would help make the layers more clear, please let us know.

**Comment by StephenOTT** [comment]
As a build on this:

http://docs.oasis-open.org/cti/stix/v2.0/cs01/part1-stix-core/stix-v2.0-cs01-part1-stix-core.html#_Toc496709309

Custom properties.  Can Custom properties be applied to all Stix "things"?   Example can i extend a Data Marking? or a Cyber Object, or something that is not a SDO/SRO which you described as "STIX Objects" in 
https://github.com/oasis-tcs/cti-stix2/issues/114#issuecomment-441284250.  If yes, then the first requirement of the custom properties section becomes confusing:

> A STIX Object MAY have any number of Custom Properties.

What is a STIX Object in this context?

**Comment by StephenOTT** [comment]
> Keep in mind you are missing the Language object in your list.

@jordan2175 can you provide a link to the specific object you are referencing above?

**Comment by jordan2175** [comment]
@StephenOTT It is part of STIX 2.1.  The current release of STIX 2.1 is CSD 01.  There are a few changes in 2.1 that you will want to be mindful of, if you are building libraries.  There is also a yet to be decided change to cyber observables and Observed Data.  Hopefully the TC will figure that out soon and get CSD 02 cut.  If you are building any support for TAXII, please see TAXII 2.1 WD04 for the latest.

**Comment by StephenOTT** [comment]
Ah okay ya no problem.  When 2.1 is released will update for that.

**Comment by jordan2175** [comment]
We have cleaned up the usage of the term STIX Objects and have even provided a diagram to help people better understand the various types of STIX objects.  We have also added a chart below the list of common properties that tells which common properties are on which objects.
